sensu/executor_test.go

```diff
@@ -0,0 +1,65 @@
+package sensu
+
+import (
+	"testing"
+
+	stdCheck "github.com/upfluence/sensu-client-go/Godeps/_workspace/src/github.com/upfluence/sensu-go/sensu/check"
+	"github.com/upfluence/sensu-client-go/sensu/check"
+	"github.com/upfluence/sensu-client-go/sensu/handler"
+)
+
+func validateCheckOutput(checkRequest *stdCheck.CheckRequest, expectedOutput *stdCheck.CheckOutput, t *testing.T) {
+	output, err := executeCheck(checkRequest)
+
+	if err != nil {
+		t.Fatalf("Expected error to be nil but got \"%s\" instead!", err)
+	}
+
+	if output == nil {
+		t.Fatalf("Expected non-nil output but got nil instead!")
+	}
+
+	if output.Status != expectedOutput.Status {
+		t.Errorf("Expected status to be \"%d\" but got \"%d\" instead!", expectedOutput.Status, output.Status)
+	}
+
+	if output.Output != expectedOutput.Output {
+		t.Errorf("Expected output to be \"%d\" but got \"%d\" instead!", expectedOutput.Output, output.Output)
+	}
+}
+
+func TestExecuteExtensionCheck(t *testing.T) {
+	check.Store["extension_check"] = &check.ExtensionCheck{Function: func() check.ExtensionCheckResult {
```

Comment:
I was looking at those because I dislike duplication, but I'm not sure how to use it cleanly when more than a few things change around...

I took a stab at it in ee97c4f, but it's slightly less readable. Please let me know what you think.